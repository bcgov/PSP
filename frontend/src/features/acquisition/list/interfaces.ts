export interface IAcquisitionFilter {
  acquisitionFileStatusTypeCode: string;
  acquisitionFileNameOrNumber: string;
  projectNameOrNumber: string;
}
