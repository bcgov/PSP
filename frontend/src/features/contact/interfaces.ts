export interface IContactFilter {
  summary: string;
  municipality: string;
  activeContactsOnly: boolean;
  searchBy: string;
}
