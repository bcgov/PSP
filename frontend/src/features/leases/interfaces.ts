export interface ILeaseFilter {
  pinOrPid: string;
  lFileNo: string;
  searchBy: string;
  tenantName: string;
  programs: string[];
}
