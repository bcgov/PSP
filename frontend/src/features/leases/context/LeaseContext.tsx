import { ILease } from 'interfaces';
import { noop } from 'lodash';
import * as React from 'react';
import { useState } from 'react';

export interface ILeaseState {
  lease?: ILease;
  setLease: (lease: ILease) => void;
}

export const LeaseStateContext = React.createContext<ILeaseState>({
  lease: undefined,
  setLease: noop,
});

export const LeaseContextProvider = (props: { children?: any }) => {
  const [lease, setLease] = useState<ILease>();

  return (
    <LeaseStateContext.Provider value={{ lease, setLease }}>
      {props.children}
    </LeaseStateContext.Provider>
  );
};
