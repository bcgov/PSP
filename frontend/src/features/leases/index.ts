export * from './add/AddLeaseContainer';
export * from './add/AddLeaseContainer';
export * from './add/AddLeaseLayout';
export * from './add/PropertyRow';
export * from './detail/BackToSearchButton';
export * from './detail/LeaseBreadCrumb';
export * from './detail/LeaseContainer';
export * from './detail/LeaseHeader';
export * from './detail/LeaseIndex';
export * from './detail/LeaseLayout';
export * from './detail/LeasePages/details/DetailAdministration';
export * from './detail/LeasePages/details/DetailDescription';
export * from './detail/LeasePages/details/DetailNotes';
export * from './detail/LeasePages/details/Details';
export * from './detail/LeasePages/details/DetailTermInformation';
export * from './detail/LeasePages/details/DetailTermInformationBox';
export * from './detail/LeasePages/details/DetailTerms';
export * from './detail/LeasePages/details/PropertiesInformation';
export * from './detail/LeasePages/details/PropertyInformation';
export * from './detail/LeasePages/LeasePageForm';
export * from './detail/LeaseRouter';
export * from './detail/LeaseStatusSummary';
export * from './detail/StackedPidTenantFields';
export * from './hooks/useLeaseDetail';
export * from './interfaces';
export * from './list/LeaseListView';
