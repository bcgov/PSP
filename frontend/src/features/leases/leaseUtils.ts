import { ILease } from 'interfaces';

/**
 * return all of the person tenant names and organization tenant names of this lease
 * @param lease
 */
export const getAllNames = (lease?: ILease) => {
  const allNames = (lease?.persons?.map(p => p.fullName) ?? []).concat(
    lease?.organizations?.map(p => p.name) ?? [],
  );
  return allNames.join(', ');
};
