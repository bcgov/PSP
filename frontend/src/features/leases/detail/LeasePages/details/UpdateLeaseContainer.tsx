import GenericModal from 'components/common/GenericModal';
import { LeaseStateContext } from 'features/leases/context/LeaseContext';
import { useUpdateLease } from 'features/leases/hooks/useUpdateLease';
import { addFormLeaseToApiLease, apiLeaseToAddFormLease } from 'features/leases/leaseUtils';
import { FormikProps } from 'formik/dist/types';
import { IAddFormLease, ILease } from 'interfaces';
import * as React from 'react';
import { useState } from 'react';
import { useContext, useMemo } from 'react';
import { useRef } from 'react';
import { useHistory } from 'react-router-dom';

import { UpdateLeaseForm } from './UpdateLeaseForm';

interface IAddLeaseParams {
  lease?: ILease;
  userOverride?: string;
}

export const UpdateLeaseContainer: React.FunctionComponent = props => {
  const { lease, setLease } = useContext(LeaseStateContext);
  const [addLeaseParams, setAddLeaseParams] = useState<IAddLeaseParams | undefined>();
  const { updateLease } = useUpdateLease();
  const history = useHistory();
  const addFormLease = useMemo(() => apiLeaseToAddFormLease(lease), [lease]);
  const formikRef = useRef<FormikProps<IAddFormLease>>(null);

  const onSubmit = async (lease: IAddFormLease) => {
    try {
      const leaseToUpdate = addFormLeaseToApiLease(lease);
      const updatedLease = await updateLease(leaseToUpdate, (userOverrideMessage?: string) =>
        setAddLeaseParams({ lease: leaseToUpdate, userOverride: userOverrideMessage }),
      );
      afterSubmit(updatedLease);
    } finally {
      formikRef?.current?.setSubmitting(false);
    }
  };

  const afterSubmit = (updatedLease?: ILease) => {
    if (!!updatedLease?.id) {
      formikRef?.current?.resetForm({ values: apiLeaseToAddFormLease(updatedLease) });
      setLease(updatedLease);
      history.push(`/lease/${updatedLease?.id}`);
    }
  };

  return (
    <>
      <UpdateLeaseForm
        onCancel={() => history.push(`/lease/${lease?.id}`)}
        onSubmit={onSubmit}
        initialValues={addFormLease}
        formikRef={formikRef}
      />
      <GenericModal
        title="Warning"
        display={!!addLeaseParams}
        message={addLeaseParams?.userOverride}
        handleOk={async () => {
          if (!!addLeaseParams?.lease) {
            const leaseResponse = await updateLease(addLeaseParams.lease, undefined, true);
            afterSubmit(leaseResponse);
            setAddLeaseParams(undefined);
          }
        }}
        handleCancel={() => setAddLeaseParams(undefined)}
        okButtonText="Save Anyways"
        okButtonVariant="warning"
        cancelButtonText="Cancel"
      />
    </>
  );
};

export default UpdateLeaseContainer;
