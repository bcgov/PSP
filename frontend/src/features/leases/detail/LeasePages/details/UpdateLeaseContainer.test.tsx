import userEvent from '@testing-library/user-event';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { IAddLeaseContainerProps } from 'features/leases';
import { LeaseStateContext } from 'features/leases/context/LeaseContext';
import { createMemoryHistory } from 'history';
import { defaultLease } from 'interfaces';
import { noop } from 'lodash';
import { mockLookups } from 'mocks/mockLookups';
import React from 'react';
import { lookupCodesSlice } from 'store/slices/lookupCodes';
import { fillInput, renderAsync, RenderOptions, waitFor } from 'utils/test-utils';

import { UpdateLeaseContainer } from './UpdateLeaseContainer';

const history = createMemoryHistory();
const storeState = {
  [lookupCodesSlice.name]: { lookupCodes: mockLookups },
};
const mockAxios = new MockAdapter(axios);

describe('Update lease container component', () => {
  const setup = async (renderOptions: RenderOptions & Partial<IAddLeaseContainerProps> = {}) => {
    // render component under test
    const component = await renderAsync(
      <LeaseStateContext.Provider value={{ lease: { ...defaultLease, id: 1 }, setLease: noop }}>
        <UpdateLeaseContainer />
      </LeaseStateContext.Provider>,
      {
        ...renderOptions,
        store: storeState,
        history,
      },
    );

    return {
      component,
    };
  };

  beforeEach(() => {
    mockAxios.resetHistory();
  });
  it('renders as expected', async () => {
    const { component } = await setup({});
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('cancels the form', async () => {
    const {
      component: { getAllByText },
    } = await setup({});
    userEvent.click(getAllByText('Cancel')[0]);
    expect(history.location.pathname).toBe('/lease/1');
  });

  it('saves the form with minimal data', async () => {
    const {
      component: { getByText, container },
    } = await setup({});
    await fillInput(container, 'startDate', '01/01/2020', 'datepicker');
    await fillInput(container, 'expiryDate', '01/02/2020', 'datepicker');
    await fillInput(container, 'paymentReceivableType', 'RCVBL', 'select');
    await fillInput(container, 'region', '1', 'select');
    await fillInput(container, 'programType', 'BCFERRIES', 'select');
    await fillInput(container, 'type', 'LICONSTRC', 'select');
    await fillInput(container, 'purposeType', 'BCFERRIES', 'select');

    mockAxios.onPut().reply(200, {});
    userEvent.click(getByText('Save'));
    await waitFor(() => {
      expect(mockAxios.history.put[0].data).toEqual(expectedFormData);
    });
  });

  it('triggers the confirm popup', async () => {
    const {
      component: { getByText, findByText, container },
    } = await setup({});
    await fillInput(container, 'startDate', '01/01/2020', 'datepicker');
    await fillInput(container, 'expiryDate', '01/02/2020', 'datepicker');
    await fillInput(container, 'paymentReceivableType', 'RCVBL', 'select');
    await fillInput(container, 'region', '1', 'select');
    await fillInput(container, 'programType', 'BCFERRIES', 'select');
    await fillInput(container, 'type', 'LICONSTRC', 'select');
    await fillInput(container, 'purposeType', 'BCFERRIES', 'select');

    mockAxios.onPut().reply(409, { error: 'test message' });
    userEvent.click(getByText('Save'));
    expect(await findByText('test message')).toBeVisible();
  });

  it('clicking on the save anyways popup saves the form', async () => {
    const {
      component: { getByText, findByText, container },
    } = await setup({});
    await fillInput(container, 'startDate', '01/01/2020', 'datepicker');
    await fillInput(container, 'expiryDate', '01/02/2020', 'datepicker');
    await fillInput(container, 'paymentReceivableType', 'RCVBL', 'select');
    await fillInput(container, 'region', '1', 'select');
    await fillInput(container, 'programType', 'BCFERRIES', 'select');
    await fillInput(container, 'type', 'LICONSTRC', 'select');
    await fillInput(container, 'purposeType', 'BCFERRIES', 'select');

    mockAxios.onPut().reply(409, { error: 'test message' });
    userEvent.click(getByText('Save'));
    userEvent.click(await findByText('Save Anyways'));
    await waitFor(() => {
      expect(mockAxios.history.put[1].data).toEqual(expectedFormData);
    });
  });
});

const expectedFormData =
  '{"lFileNo":"","tfaFileNo":0,"expiryDate":"2020-01-02","startDate":"2020-01-01","paymentReceivableType":{"id":"RCVBL"},"purposeType":{"id":"BCFERRIES"},"responsibilityType":{"id":"HQ"},"initiatorType":{"id":"PROJECT"},"type":{"id":"LICONSTRC"},"statusType":{"id":"ACTIVE"},"region":{"regionCode":"1"},"programType":{"id":"BCFERRIES"},"otherType":"","otherProgramType":"","otherCategoryType":"","otherPurposeType":"","note":"","motiName":"Moti, Name, Name","amount":0,"renewalCount":0,"description":"","landArea":"","areaUnit":"","isResidential":false,"isCommercialBuilding":false,"isOtherImprovement":false,"returnNotes":"","documentationReference":"","tenantNotes":[],"insurances":[],"terms":[],"tenants":[],"properties":[],"persons":[],"organizations":[],"improvements":[],"securityDeposits":[],"securityDepositReturns":[],"id":1}';
