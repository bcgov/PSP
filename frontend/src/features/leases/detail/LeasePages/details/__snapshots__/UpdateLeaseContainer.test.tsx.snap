// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update lease container component renders as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div
    class="Toastify"
  />
  .c3.c3.form-control.is-valid {
  background-image: none;
}

.c3.c3.form-control.is-invalid {
  border-color: #d8292f !important;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  gap: 0.8rem;
}

.c7 .form-label {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  gap: 0.8rem;
}

.c6 .form-label {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline !important;
  -webkit-box-align: baseline !important;
  -ms-flex-align: baseline !important;
  align-items: baseline !important;
  gap: 2rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  gap: 0.8rem;
}

.c9 .form-label {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c8 input.form-control {
  min-width: 50rem;
  max-width: 50rem;
}

.c10 textarea.form-control {
  min-width: 80rem;
  height: 7rem;
  resize: none;
}

.c11 textarea.form-control {
  min-width: 80rem;
  height: 18rem;
  width: 80rem;
  resize: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  padding: 0 2.5rem;
  height: 100%;
  overflow-y: auto;
  text-align: left;
}

.c0 .required .form-label:after {
  content: ' *';
}

.c0 .row {
  padding: 0.5rem 0;
}

.c0 .form-label {
  font-weight: 700;
}

.c0 .row .col:first-of-type .form-label {
  min-width: 13.5rem;
}

.c0 .form-control {
  min-width: 25rem;
  max-width: 25rem;
}

.c0 .col {
  -webkit-align-items: end;
  -webkit-box-align: end;
  -ms-flex-align: end;
  align-items: end;
  width: 100%;
}

.c5 {
  font-family: 'BcSans-Bold';
  font-size: 2rem;
  margin-bottom: 1rem;
  text-align: left;
  padding: 1rem 0 0.5rem 0;
  border-bottom: solid 0.3rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

<form
    class="c0"
    novalidate=""
  >
    <div
      class="row"
    >
      <div
        class="c1 col"
      >
        <div
          class="required c2 form-group"
        >
          <label
            class="form-label"
            for="datepicker-startDate"
          >
            Start Date
          </label>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                autocomplete="off"
                class="c3 form-control date-picker"
                id="datepicker-startDate"
                name="startDate"
                placeholder="--/--/----"
                required=""
                type="text"
                value="01/01/2020"
              />
            </div>
          </div>
        </div>
        <div
          class="c2 form-group"
        >
          <label
            class="form-label"
            for="datepicker-expiryDate"
          >
            Expiry Date (if known)
          </label>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                autocomplete="off"
                class="c3 form-control date-picker"
                id="datepicker-expiryDate"
                name="expiryDate"
                placeholder="--/--/----"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="c4"
        />
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="input-statusType"
          >
            Status: 
          </label>
          <select
            class="form-select form-control"
            id="input-statusType"
            name="statusType"
          >
            <option
              value=""
            >
              Select Status
            </option>
            <option
              class="option"
              value="ACTIVE"
            >
              Active
            </option>
            <option
              class="option"
              value="DISCARD"
            >
              Discarded
            </option>
            <option
              class="option"
              value="DRAFT"
            >
              Draft
            </option>
            <option
              class="option"
              value="EXPIRED"
            >
              Expired
            </option>
            <option
              class="option"
              value="TERMINATED"
            >
              Terminated
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h3
          class="c5"
        >
          Administration
        </h3>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="required c6 form-group"
        >
          <label
            class="form-label"
            for="input-paymentReceivableType"
          >
            Receivable or Payable: 
          </label>
          <select
            class="form-select form-control"
            id="input-paymentReceivableType"
            name="paymentReceivableType"
            required=""
          >
            <option
              value=""
            >
              Select
            </option>
            <option
              class="option"
              value="PYBLBCTFA"
            >
              Payable (BCTFA as tenant)
            </option>
            <option
              class="option"
              value="PYBLMOTI"
            >
              Payable (MOTI as tenant)
            </option>
            <option
              class="option"
              value="RCVBL"
            >
              Receivable
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="c7 c8 input form-group"
        >
          <label
            class="form-label"
            for="input-motiName"
          >
            MOTI Contact: 
          </label>
          <input
            class="form-control"
            id="input-motiName"
            name="motiName"
            value="Moti, Name, Name"
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="required c6 form-group"
        >
          <label
            class="form-label"
            for="input-region"
          >
            MOTI Region: 
          </label>
          <select
            class="form-select form-control"
            id="input-region"
            name="region"
            required=""
          >
            <option
              value=""
            >
              Select region
            </option>
            <option
              class="option"
              value="1"
            >
              South Coast Region
            </option>
            <option
              class="option"
              value="2"
            >
              Southern Interior Region
            </option>
            <option
              class="option"
              value="3"
            >
              Northern Region
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="c1 col"
      >
        <div
          class="required c6 form-group"
        >
          <label
            class="form-label"
            for="input-programType"
          >
            Program: 
          </label>
          <select
            class="form-select form-control"
            id="input-programType"
            name="programType"
            required=""
          >
            <option
              value=""
            >
              Select program
            </option>
            <option
              class="option"
              value="BCFERRIES"
            >
              BC Ferries
            </option>
            <option
              class="option"
              value="BCTRANSIT"
            >
              BC Transit
            </option>
            <option
              class="option"
              value="BELLETERM"
            >
              Belleville Terminal
            </option>
            <option
              class="option"
              value="COMMBLDG"
            >
              Commercial Buildings
            </option>
            <option
              class="option"
              value="LCLGOVT"
            >
              Local Government/RCMP
            </option>
            <option
              class="option"
              value="OTHER"
            >
              Other Licencing
            </option>
            <option
              class="option"
              value="RAILTRAIL"
            >
              Rail Trails
            </option>
            <option
              class="option"
              value="RESRENTAL"
            >
              Residential Rentals
            </option>
            <option
              class="option"
              value="TMEP"
            >
              TMEP
            </option>
            <option
              class="option"
              value="TRANSLINK"
            >
              TransLink
            </option>
            <option
              class="option"
              value="UTILITY"
            >
              Utilities
            </option>
          </select>
        </div>
        <div
          class="required c7 input form-group"
        >
          <label
            class="form-label"
            for="input-otherProgramType"
          >
            Other Program: 
          </label>
          <input
            class="form-control"
            id="input-otherProgramType"
            name="otherProgramType"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="c1 col"
      >
        <div
          class="required c6 form-group"
        >
          <label
            class="form-label"
            for="input-type"
          >
            Type: 
          </label>
          <select
            class="form-select form-control"
            id="input-type"
            name="type"
            required=""
          >
            <option
              value=""
            >
              Select type
            </option>
            <option
              class="option"
              value="LICONSTRC"
            >
              License to Construct
            </option>
            <option
              class="option"
              value="LIMOTIPRJ"
            >
              MOTI Project Use License
            </option>
            <option
              class="option"
              value="LIOCCACCS"
            >
              License of Occupation (access)
            </option>
            <option
              class="option"
              value="LIOCCTTLD"
            >
              License of Occupation (titled)
            </option>
            <option
              class="option"
              value="LIOCCUSE"
            >
              License of Occupation (use)
            </option>
            <option
              class="option"
              value="LIOCCUTIL"
            >
              License of Occupation (utilities)
            </option>
            <option
              class="option"
              value="LIPPUBHWY"
            >
              License of Prov Public Highway
            </option>
            <option
              class="option"
              value="LSGRND"
            >
              Ground Lease
            </option>
            <option
              class="option"
              value="LSREG"
            >
              Lease - Registered
            </option>
            <option
              class="option"
              value="LSUNREG"
            >
              Lease - Unregistered
            </option>
            <option
              class="option"
              value="MANUFHOME"
            >
              Manufactured Home Act
            </option>
            <option
              class="option"
              value="OTHER"
            >
              Other
            </option>
            <option
              class="option"
              value="RESLNDTEN"
            >
              Residential Tenancy Act
            </option>
            <option
              class="option"
              value="ROADXING"
            >
              Road Crossing
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="c1 col"
      >
        <div
          class="required c6 form-group"
        >
          <label
            class="form-label"
            for="input-categoryType"
          >
            Category: 
          </label>
          <select
            class="form-select form-control"
            id="input-categoryType"
            name="categoryType"
            required=""
          >
            <option
              value=""
            >
              Select category
            </option>
            <option
              class="option"
              value="AGRIC"
            >
              Agricultural
            </option>
            <option
              class="option"
              value="COMM"
            >
              Commercial
            </option>
            <option
              class="option"
              value="GOVGOV"
            >
              Gov't to Gov't
            </option>
            <option
              class="option"
              value="INDUS"
            >
              Industrial
            </option>
            <option
              class="option"
              value="OTHER"
            >
              Other
            </option>
            <option
              class="option"
              value="RESID"
            >
              Residential
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="c1 col"
      >
        <div
          class="required c6 form-group"
        >
          <label
            class="form-label"
            for="input-purposeType"
          >
            Purpose: 
          </label>
          <select
            class="form-select form-control"
            id="input-purposeType"
            name="purposeType"
            required=""
          >
            <option
              value=""
            >
              Select purpose
            </option>
            <option
              class="option"
              value="ACCCCOM"
            >
              Access (Industrial/Commercial)
            </option>
            <option
              class="option"
              value="ACCRES"
            >
              Access (Residential)
            </option>
            <option
              class="option"
              value="AGRIC"
            >
              Agricultural
            </option>
            <option
              class="option"
              value="BCFERRIES"
            >
              BC Ferries
            </option>
            <option
              class="option"
              value="CAMPING"
            >
              Camping
            </option>
            <option
              class="option"
              value="COMMBLDG"
            >
              Commercial Building
            </option>
            <option
              class="option"
              value="EMERGSVCS"
            >
              Emergency Services
            </option>
            <option
              class="option"
              value="ENCROACH"
            >
              Encroachment
            </option>
            <option
              class="option"
              value="FENCEGATE"
            >
              Fencing/Gate
            </option>
            <option
              class="option"
              value="GARDENING"
            >
              Gardening
            </option>
            <option
              class="option"
              value="GRAVEL"
            >
              Gravel
            </option>
            <option
              class="option"
              value="GRAZING"
            >
              Grazing
            </option>
            <option
              class="option"
              value="LOGGING"
            >
              Logging/Timber Harvest
            </option>
            <option
              class="option"
              value="MARINEFAC"
            >
              Marine Facility
            </option>
            <option
              class="option"
              value="MOBILEHM"
            >
              Mobile Home Pad
            </option>
            <option
              class="option"
              value="MTCYARD"
            >
              Maintenance Yard
            </option>
            <option
              class="option"
              value="OTHER"
            >
              Other*
            </option>
            <option
              class="option"
              value="PARK"
            >
              Park
            </option>
            <option
              class="option"
              value="PARKING"
            >
              Parking
            </option>
            <option
              class="option"
              value="PARKNRID"
            >
              Park and Ride
            </option>
            <option
              class="option"
              value="PIPELINE"
            >
              Pipeline
            </option>
            <option
              class="option"
              value="PRELOAD"
            >
              Preload
            </option>
            <option
              class="option"
              value="PRVTRANS"
            >
              Private transportation
            </option>
            <option
              class="option"
              value="RAILWAY"
            >
              Railway
            </option>
            <option
              class="option"
              value="RESTAREA"
            >
              Rest Area/Pullout
            </option>
            <option
              class="option"
              value="SIDEWALK"
            >
              Sidewalks/Landscaping
            </option>
            <option
              class="option"
              value="SIGNAGE"
            >
              Signage
            </option>
            <option
              class="option"
              value="SPCLEVNT"
            >
              Special Event
            </option>
            <option
              class="option"
              value="STGNGAREA"
            >
              Staging Area
            </option>
            <option
              class="option"
              value="STKPILING"
            >
              Stockpiling
            </option>
            <option
              class="option"
              value="STORAGE"
            >
              Storage
            </option>
            <option
              class="option"
              value="TOURINFO"
            >
              Tourist Info
            </option>
            <option
              class="option"
              value="TRAIL"
            >
              Trail
            </option>
            <option
              class="option"
              value="UTILINFRA"
            >
              Utilities-infrastructure
            </option>
            <option
              class="option"
              value="UTILOHDXG"
            >
              Utilities-overhead xing
            </option>
            <option
              class="option"
              value="UTILUGDXG"
            >
              Utilities-underground xing
            </option>
            <option
              class="option"
              value="WATERRES"
            >
              Water Reservoir
            </option>
            <option
              class="option"
              value="WEIGHSCL"
            >
              Weigh Scale
            </option>
            <option
              class="option"
              value="XING"
            >
              Crossing
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="c6 form-group"
        >
          <label
            class="form-label"
            for="input-initiatorType"
          >
            Initiator: 
            <span
              class="tooltip-icon"
              data-testid="tooltip-icon"
            >
              <svg
                class="tooltip-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                />
              </svg>
            </span>
          </label>
          <select
            class="form-select form-control"
            id="input-initiatorType"
            name="initiatorType"
          >
            <option
              value=""
            >
              Select initiator
            </option>
            <option
              class="option"
              value="HQ"
            >
              Headquarters
            </option>
            <option
              class="option"
              value="PROJECT"
            >
              Project
            </option>
            <option
              class="option"
              value="REGION"
            >
              Region
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="c1 col"
      >
        <div
          class="c6 form-group"
        >
          <label
            class="form-label"
            for="input-responsibilityType"
          >
            Responsibility: 
            <span
              class="tooltip-icon"
              data-testid="tooltip-icon"
            >
              <svg
                class="tooltip-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                />
              </svg>
            </span>
          </label>
          <select
            class="form-select form-control"
            id="input-responsibilityType"
            name="responsibilityType"
          >
            <option
              value=""
            >
              Select group responsible
            </option>
            <option
              class="option"
              value="HQ"
            >
              Headquarters
            </option>
            <option
              class="option"
              value="PROJECT"
            >
              Project
            </option>
            <option
              class="option"
              value="REGION"
            >
              Region
            </option>
          </select>
        </div>
        <div
          class="c9 form-group"
        >
          <label
            class="form-label"
            for="datepicker-responsibilityEffectiveDate"
          >
            Effective date of responsibility:
          </label>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                autocomplete="off"
                class="c3 form-control date-picker"
                id="datepicker-responsibilityEffectiveDate"
                name="responsibilityEffectiveDate"
                placeholder="--/--/----"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h3
          class="c5"
        >
          Reference Information
        </h3>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="c10 textarea input form-group"
        >
          <label
            class="form-label"
            for="input-documentationReference"
          >
            Location of documents: 
            <span
              class="tooltip-icon"
              data-testid="tooltip-icon"
            >
              <svg
                class="tooltip-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                />
              </svg>
            </span>
          </label>
          <textarea
            class="form-control"
            id="input-documentationReference"
            name="documentationReference"
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="c7 c8 input form-group"
        >
          <label
            class="form-label"
            for="input-tfaFileNo"
          >
            LIS # 
          </label>
          <input
            class="form-control"
            id="input-tfaFileNo"
            name="tfaFileNo"
            type="number"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="c7 c8 input form-group"
        >
          <label
            class="form-label"
            for="input-psFileNo"
          >
            PS #: 
          </label>
          <input
            class="form-control"
            id="input-psFileNo"
            name="psFileNo"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <h3
          class="c5"
        >
          Property Information
        </h3>
      </div>
    </div>
    <button
      class="btn btn-link"
      type="button"
    >
      + Add another property
    </button>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="c11 textarea input form-group"
        >
          <label
            class="form-label"
            for="input-description"
          >
            Description 
          </label>
          <textarea
            class="form-control"
            id="input-description"
            name="description"
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="c11 textarea input form-group"
        >
          <label
            class="form-label"
            for="input-note"
          >
            Notes 
          </label>
          <textarea
            class="form-control"
            id="input-note"
            name="note"
          />
        </div>
      </div>
    </div>
  </form>
  .c0 {
  z-index: 100;
  width: 100%;
  padding: 0.5rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  gap: 1rem;
  position: -webkit-sticky;
  position: sticky;
  bottom: 0rem;
  background-color: white;
}

.c0 .btn {
  min-height: 3rem;
}

<div
    class="c0"
  >
    <button
      class="Button btn btn-secondary"
      type="button"
    >
      <div
        class="Button__value"
      >
        Cancel
      </div>
    </button>
    <button
      class="Button Button--disabled btn btn-primary"
      disabled=""
      type="button"
    >
      <div
        class="Button__value"
      >
        Save
      </div>
    </button>
  </div>
  <div
    class="container"
  />
</DocumentFragment>
`;
